javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java

```diff
@@ -556,4 +559,22 @@ public static Name getElementValueClassName(AnnotationMirror anno, CharSequence
         }
         map.put(key, Collections.unmodifiableSet(result));
     }
+
+    /**
+     * Returns the annotations explicitly written on a constructor result.
+     * Callers should check that {@code constructorDeclaration} is in fact a declaration
+     * of a constructor.
+     *
+     * @param constructorDeclaration Declaration tree of constructor
+     * @return set of annotations explicit on the resulting type of the constructor
+     */
+    public static Set<AnnotationMirror> getExplicitAnnotationsOnConstructorResult(MethodTree constructorDeclaration) {
```

Comment:
Would it make sense to move this to https://github.com/typetools/checker-framework/blob/master/framework/src/org/checkerframework/framework/util/ConstructorReturnUtil.java ?
That class only has a single method and seems rather underused.